src/Microsoft.OData.Core/UriParser/Resolver/ODataUriResolver.cs

```diff
@@ -127,6 +128,33 @@ public virtual IEdmProperty ResolveProperty(IEdmStructuredType type, string prop
             return type.FindProperty(propertyName);
         }
 
+        /// <summary>
+        /// Resolve term name from model.
+        /// </summary>
+        /// <param name="model">The model to be used.</param>
+        /// <param name="termName">The term name to be resolved.</param>
+        /// <returns>Resolved term.</returns>
+        public virtual IEdmTerm ResolveTerm(IEdmModel model, string termName)
+        {
+            if (EnableCaseInsensitive)
+            {
+                var result = model.SchemaElements.OfType<IEdmTerm>()
+               .Where(_ => string.Equals(termName, _.FullName(), StringComparison.OrdinalIgnoreCase))
+               .ToList();
+
+                if (result.Count == 1)
+                {
+                    return result.Single();
+                }
+                else if (result.Count > 1)
+                {
+                    throw new ODataException(Strings.UriParserMetadata_MultipleMatchingTypesFound(termName));
+                }

```

Comment:
should we add returning with null here if result.Count==0?

Reply:
simplified logic in latest pr

---
In reply to: [167038073](https://github.com/OData/odata.net/pull/1063#discussion_r167038073) [](ancestors = 167038073)